src/pretty.rs

```diff
@@ -0,0 +1,471 @@
+        if let Some(types) = &mv.types {
+            self.text(": ")
+                .append(types.types.clone().pretty(self))
+                .append(self.line())
+        } else {
+            self.nil()
+        }
+        .append(
+            self.intersperse(
+                mv.contracts
+                    .iter()
+                    .map(|c| self.text("| ").append(c.to_owned().types.pretty(self))),
+                self.line().clone(),
+            ),
+        )
+        .append(if mv.priority == crate::term::MergePriority::Default {
+            self.line().append(self.text("| default"))
+        } else {
+            // if last field is not default, it will add a optional line break before the `=` sign
+            self.nil()
+        })
+        .nest(2)
+        .group()
+    }
+
+    fn atom(&'a self, rt: &RichTerm) -> DocBuilder<'a, Self, A> {
+        if rt.as_ref().is_atom() {
+            rt.to_owned().pretty(self)
+        } else {
+            rt.to_owned().pretty(self).parens()
+        }
+    }
+}
+
+impl<'a, D, A> Pretty<'a, D, A> for BinaryOp
+where
+    D: NickelAllocatorExt<'a, A>,
+    D::Doc: Clone,
+    A: Clone + 'a,
+{
+    fn pretty(self, allocator: &'a D) -> DocBuilder<'a, D, A> {
+        use BinaryOp::*;
+        match self {
+            Plus() => allocator.text("+"),
+            Sub() => allocator.text("-"),
+
+            Mult() => allocator.text("*"),
+            Div() => allocator.text("/"),
+            Modulo() => allocator.text("%"),
+
+            Eq() => allocator.text("=="),
+            LessThan() => allocator.text("<"),
+            GreaterThan() => allocator.text(">"),
+            GreaterOrEq() => allocator.text(">="),
+            LessOrEq() => allocator.text("<="),
+
+            Merge() => allocator.text("&"),
+
+            op => allocator.as_string(format!("%{:?}%", op).to_lowercase()),
+        }
+    }
+}
+
+impl<'a, D, A> Pretty<'a, D, A> for RichTerm
+where
+    D: NickelAllocatorExt<'a, A>,
+    D::Doc: Clone,
+    A: Clone + 'a,
+{
+    fn pretty(self, allocator: &'a D) -> DocBuilder<'a, D, A> {
+        use Term::*;
+
+        match self.as_ref() {
+            Null => allocator.text("null"),
+            Bool(v) => allocator.as_string(v),
+            Num(v) => allocator.as_string(v),
+            Str(v) => allocator.as_string(v),
+            StrChunks(chunks) => allocator
+                .text("m%\"")
+                .append(allocator.intersperse(
+                    chunks.iter().map(|c| {
+                        match c {
+                            crate::term::StrChunk::Literal(s) => allocator.as_string(s),
+                            crate::term::StrChunk::Expr(e, i) => allocator
+                                .text("%{")
+                                .append(e.to_owned().pretty(allocator))
+                                .append(allocator.text("}")),
+                        }
+                    }),
+                    allocator.line_(),
```

Comment:
You don't want to insert new lines that don't exists inside strings, as that changes the semantics.